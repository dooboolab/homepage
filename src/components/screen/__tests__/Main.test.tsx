import 'react-native';

import React, {ReactElement} from 'react';
import {RenderAPI, render} from '@testing-library/react-native';
import {createTestElement, createTestProps} from '../../../../test/testUtils';

import Screen from '../Main';

let props: any;
let component: ReactElement;
let testingLib: RenderAPI;

describe('Render', () => {
  beforeEach(() => {
    props = createTestProps();
    component = createTestElement(<Screen {...props} />);
    testingLib = render(component);
  });

  it('renders without crashing', () => {
    const baseElement = testingLib.toJSON();

    expect(baseElement).toMatchSnapshot();
    expect(baseElement).toBeTruthy();
  });
});
