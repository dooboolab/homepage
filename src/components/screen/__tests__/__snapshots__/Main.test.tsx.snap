// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render renders without crashing 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "flex-start",
          "alignSelf": "stretch",
          "backgroundColor": "#FFFFFF",
          "flexBasis": 0,
          "flexDirection": "column",
          "flexGrow": 1,
          "flexShrink": 1,
          "justifyContent": "flex-start",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "column",
            "flexWrap": "wrap",
            "justifyContent": "center",
            "paddingBottom": 0,
            "paddingLeft": 24,
            "paddingRight": 24,
            "paddingTop": 0,
            "width": "100%",
          },
        ]
      }
    >
      <Image
        source={
          Object {
            "canInstrument": true,
            "createTransformer": [Function],
            "getCacheKey": [Function],
            "process": [Function],
          }
        }
        style={
          Array [
            Object {
              "height": 66.54,
              "marginTop": 8,
              "width": 180,
            },
          ]
        }
      />
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexDirection": "row",
            },
          ]
        }
      >
        <View
          onPress={[Function]}
          style={
            Array [
              Object {
                "alignItems": "center",
                "justifyContent": "center",
              },
              Object {
                "marginRight": 20,
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "fontSize": 18,
                  "fontWeight": "500",
                  "paddingBottom": 20,
                  "paddingLeft": 20,
                  "paddingRight": 20,
                  "paddingTop": 20,
                },
              ]
            }
          >
            Story
          </Text>
        </View>
        <View
          onPress={[Function]}
          style={
            Array [
              Object {
                "alignItems": "center",
                "justifyContent": "center",
              },
              Object {
                "marginRight": 20,
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "fontSize": 18,
                  "fontWeight": "500",
                  "paddingBottom": 20,
                  "paddingLeft": 20,
                  "paddingRight": 20,
                  "paddingTop": 20,
                },
              ]
            }
          >
            Work
          </Text>
        </View>
        <View
          onPress={[Function]}
          style={
            Array [
              Object {
                "alignItems": "center",
                "justifyContent": "center",
              },
              Object {
                "marginRight": 20,
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "fontSize": 18,
                  "fontWeight": "500",
                  "paddingBottom": 20,
                  "paddingLeft": 20,
                  "paddingRight": 20,
                  "paddingTop": 20,
                },
              ]
            }
          >
            Contact
          </Text>
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "position": "absolute",
              "right": 20,
              "top": 20,
            },
          ]
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
            }
          }
        >
          <View
            activeOpacity={0.8}
            onPress={[Function]}
            style={
              Array [
                Object {
                  "backgroundColor": "#ecf0f1",
                  "borderRadius": 20,
                  "justifyContent": "center",
                  "padding": 12,
                  "width": 46,
                },
                Object {},
              ]
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "white",
                  "borderRadius": 9,
                  "elevation": 1.5,
                  "height": 18,
                  "justifyContent": "center",
                  "left": 0,
                  "margin": 4,
                  "position": "absolute",
                  "shadowColor": "#000",
                  "shadowOffset": Object {
                    "height": 2,
                    "width": 0,
                  },
                  "shadowOpacity": 0.2,
                  "shadowRadius": 2.5,
                  "transform": Array [
                    Object {
                      "translateX": 0,
                    },
                  ],
                  "width": 18,
                }
              }
            />
          </View>
        </View>
      </View>
    </View>
    <Text
      style={
        Array [
          Object {
            "color": "#000000",
          },
        ]
      }
    >
      Main
    </Text>
  </View>
</View>
`;
